File: xsolla-login-sdk/src/main/java/com/xsolla/android/login/entity/response/AuthResponse.java
package com.xsolla.android.login.entity.response;

import com.google.gson.annotations.SerializedName;
import com.xsolla.android.login.token.TokenUtils;

public class AuthResponse {

    @SerializedName("login_url")
    private String loginUrl;

    public String getToken() {
        return TokenUtils.getTokenFromUrl(loginUrl);
    }

}

File: xsolla-login-sdk/src/main/java/com/xsolla/android/login/entity/response/AuthResponse.kt
package com.xsolla.android.login.entity.response

import com.google.gson.annotations.SerializedName
import com.xsolla.android.login.token.TokenUtils

internal data class AuthResponse(
    @SerializedName("login_url")
    val loginUrl: String
) {
    fun getToken() = TokenUtils.getTokenFromUrl(loginUrl)
}